tests/chainer_tests/links_tests/model_tests/test_vision.py

```diff
@@ -0,0 +1,220 @@
+        self.assertEqual(y1.dtype, numpy.float32)
+        y2 = resnet.prepare(x2)
+        self.assertEqual(y2.shape, (3, 224, 224))
+        self.assertEqual(y2.dtype, numpy.float32)
+        y3 = resnet.prepare(x3, size=None)
+        self.assertEqual(y3.shape, (3, 160, 120))
+        self.assertEqual(y3.dtype, numpy.float32)
+        y4 = resnet.prepare(x4)
+        self.assertEqual(y4.shape, (3, 224, 224))
+        self.assertEqual(y4.dtype, numpy.float32)
+        y5 = resnet.prepare(x5, size=None)
+        self.assertEqual(y5.shape, (3, 160, 120))
+        self.assertEqual(y5.dtype, numpy.float32)
+
+    def check_extract(self):
+        x1 = numpy.random.uniform(0, 255, (320, 240, 3)).astype(numpy.uint8)
+        x2 = numpy.random.uniform(0, 255, (320, 240)).astype(numpy.uint8)
+        result = self.link.extract([x1, x2], layers=['res3', 'pool5'])
+        self.assertEqual(len(result), 2)
+        y1 = cuda.to_cpu(result['res3'].data)
+        self.assertEqual(y1.shape, (2, 512, 28, 28))
+        self.assertEqual(y1.dtype, numpy.float32)
+        y2 = cuda.to_cpu(result['pool5'].data)
+        self.assertEqual(y2.shape, (2, 2048))
+        self.assertEqual(y2.dtype, numpy.float32)
+
+        x3 = numpy.random.uniform(0, 255, (80, 60)).astype(numpy.uint8)
+        result = self.link.extract([x3], layers=['res2'], size=None)
+        self.assertEqual(len(result), 1)
+        y3 = cuda.to_cpu(result['res2'].data)
+        self.assertEqual(y3.shape, (1, 256, 20, 15))
+        self.assertEqual(y3.dtype, numpy.float32)
+
+    def test_extract_cpu(self):
+        self.check_extract()
+
+    @attr.gpu
+    def test_extract_gpu(self):
+        self.link.to_gpu()
+        self.check_extract()
+
+    def check_predict(self):
+        x1 = numpy.random.uniform(0, 255, (320, 240, 3)).astype(numpy.uint8)
+        x2 = numpy.random.uniform(0, 255, (320, 240)).astype(numpy.uint8)
+        result = self.link.predict([x1, x2], oversample=False)
+        y = cuda.to_cpu(result.data)
+        self.assertEqual(y.shape, (2, 1000))
+        self.assertEqual(y.dtype, numpy.float32)
+        result = self.link.predict([x1, x2], oversample=True)
+        y = cuda.to_cpu(result.data)
+        self.assertEqual(y.shape, (2, 1000))
+        self.assertEqual(y.dtype, numpy.float32)
+
+    def test_predict_cpu(self):
+        self.check_predict()
+
+    @attr.gpu
+    def test_predict_gpu(self):
+        self.link.to_gpu()
+        self.check_predict()
+
+
+@unittest.skipUnless(resnet.available, 'Pillow is required')
+class TestVGG16Layers(unittest.TestCase):
+
+    def setUp(self):
+        self.link = vgg.VGG16Layers(pretrained_model=None)
+
+        # Suppress warning that arises from zero division.
+        warnings.filterwarnings('ignore', category=RuntimeWarning)
+
+    def test_available_layers(self):
+        result = self.link.available_layers
+        self.assertIsInstance(result, list)
+        self.assertIsNot(len(result), 0)
```

Comment:
ditto
The list of available_layers for VGG16 is known and fixed. Can't we use the length directly here?